tree/[M]652.cpp

```diff
@@ -0,0 +1,6 @@
+// 652. Find Duplicate Subtrees - Medium
+// https://leetcode.com/problems/find-duplicate-subtrees/
+
+/* Analysis:
+- Bài này anh nghĩ phải dùng DP, nhưng đang chưa biết lưu dữ liệu kiểu gì, chú có gợi ý gì ko?
```

Comment:
2 hints for you: 1) if 2 substrees are equals, the roots must be at the same layer. 2) you can write a function to check if 2 roots are equal or not. Ex: bool checkEqual(TreeNode node1, TreeNode node2); use left and right to check.